qubes/events.py

```diff
@@ -115,6 +115,17 @@ def add_handler(cls, event, func):
         # pylint: disable=no-member
         cls.__handlers__[event].add(func)
 
+    @classmethod
+    def remove_handler(cls, event, func):
+        '''Remove event handler from subject's class.
+
+        :param str event: event identificator
+        :param collections.Callable handler: handler callable
+        '''
+
+        # pylint: disable=no-member
+        cls.__handlers__[event].remove(func)
+
```

Comment:
Yes, it somehow will. However, the current code is sequential (since the handlers are called in turns) and this code could be async-aware. And I think it should be around 10 lines, including registering and deregistering sockets.